spec/presenters/single_content_item/single_content_item_presenter_spec.rb

```diff
@@ -98,6 +98,104 @@
     end
   end
 
+  describe '#total_upviews' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:time_series_metrics] = [{ name: 'upviews', total: 500_000_000 }, { name: 'pviews', total: 0 }]
+
+      expect(subject.total_upviews).to eq('500,000,000')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:time_series_metrics] = [{ name: 'upviews', total: 5 }, { name: 'pviews', total: 0 }]
+
+      expect(subject.total_upviews).to eq('5')
+    end
+  end
+
+  describe '#total_pviews' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:time_series_metrics] = [{ name: 'pviews', total: 500_000_000 }, { name: 'upviews', total: 0 }]
+
+      expect(subject.total_pviews).to eq('500,000,000')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:time_series_metrics] = [{ name: 'pviews', total: 5 }, { name: 'upviews', total: 0 }]
+
+      expect(subject.total_pviews).to eq('5')
+    end
+  end
+
+  describe '#total_searches' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:time_series_metrics] = [{ name: 'searches', total: 500_000_000 }, { name: 'upviews', total: '0' }, { name: 'pviews', total: 0 }]
+
+      expect(subject.total_searches).to eq('500,000,000')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:time_series_metrics] = [{ name: 'searches', total: 5 }, { name: 'pviews', total: 5 }, { name: 'upviews', total: 0 }]
+
+      expect(subject.total_searches).to eq('5')
+    end
+  end
+
+  describe '#total_satisfaction' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:time_series_metrics] = [{ name: 'satisfaction', total: 1 }, { name: 'upviews', total: '0' }, { name: 'pviews', total: 0 }]
+
+      expect(subject.total_satisfaction).to eq('100%')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:time_series_metrics] = [{ name: 'satisfaction', total: 0.01 }, { name: 'pviews', total: 5 }, { name: 'upviews', total: 0 }]
+
+      expect(subject.total_satisfaction).to eq('1%')
+    end
+  end
+
+  describe '#total_feedex' do
+    it 'correctly formats number for a value in the millions' do
+      current_period_data[:time_series_metrics] = [{ name: 'feedex', total: 1_000 }, { name: 'upviews', total: '0' }, { name: 'pviews', total: 0 }]
+
+      expect(subject.total_feedex).to eq('1,000')
+    end
+
+    it 'correctly formats number for a small value' do
+      current_period_data[:time_series_metrics] = [{ name: 'feedex', total: 1 }, { name: 'pviews', total: 5 }, { name: 'upviews', total: 0 }]
```

Comment:
In this test, and some of the tests above, I think some of the test data in `current_period_data[:time_series_metrics]` is redundant, e.g. the `pviews` value here. If so, it would be good to remove it to make the tests clearer.

Reply:
There's some code in the presenter tied up to the stuff in the private section whenever the presenter is initialised (`assign_pageviews_per_visit`). It always expects these values, but I've not had a chance to refactor that area yet and looked worth tackling in another PR.